extensions/interactions/TextInput/protractor.js

```diff
@@ -27,12 +27,55 @@ var customizeInteraction = function(elem, placeholderText, heightOfBox) {
   ).setValue(heightOfBox);
 };
 
-var expectInteractionDetailsToMatch = function(elem) {
-  // TODO (Jacob) add checks for the placeholder text and the height of the
-  // textbox.
+var expectInteractionDetailsToMatch = function(elem, placeholderText,
+  heightOfBox) {
+  var textInputElem = elem.element(by.tagName('oppia-interactive-text-input'));
   expect(
-    elem.element(by.tagName('oppia-interactive-text-input')).isPresent()
+    textInputElem.isPresent()
   ).toBe(true);
+
+  var isCheckForPlaceHolder = (typeof placeholderText !== 'undefined');
+  var isCheckForBoxHeight = (typeof heightOfBox !== 'undefined');
+
+  if (isCheckForPlaceHolder || isCheckForBoxHeight) {
+    var textAreaElem = textInputElem.element(by.tagName('textarea'));
+    var inputElem = textInputElem.element(by.tagName('input'));
+
+    if (isCheckForPlaceHolder) {
+      textAreaElem.isPresent().then(function(present) {
+        if (present) {
```

Comment:
I don't understand why you need to look in two different places for the placeholder text. Is something strange going on in the DOM?